pex/testing.py

```diff
@@ -551,6 +557,122 @@ def ensure_python_interpreter(version):
     return python
 
 
+class InterpreterImplementation(Enum["InterpreterImplementation.Value"]):
+    class Value(Enum.Value):
+        pass
+
+    CPython = Value("CPython")
+    PyPy = Value("PyPy")
+
+
+def find_python_interpreter(
+    version=(),  # type: Tuple[int, ...]
+    implementation=InterpreterImplementation.CPython,  # type: InterpreterImplementation.Value
+):
+    # type: (...) -> Optional[str]
+    for interpreter in PythonInterpreter.iter():
```

Comment:
The Windows support branch will add a skip here for `version < (3, 8)` since Windows Python 3.7 doesn't have long path support enabled in its manifest and lower versions don't have symlink support. This will get Windows running all the exisitng ITs it can that do not require Python 2.7.